manual/manual/cmds/spacetime.etex

```diff
@@ -0,0 +1,118 @@
+\chapter{Memory profiling with Spacetime}
+\pdfchapterfold{-9}{Memory profiling with Spacetime}
+%HEVEA\cutname{spacetime.html}
+
+\section{Overview}
+
+Spacetime is the name given to functionality within the OCaml compiler that
+provides for accurate profiling of the memory behaviour of a program.
+Using Spacetime it is possible to determine the source of memory leaks
+and excess memory allocation quickly and easily.  Excess allocation slows
+programs down both by imposing a higher load on the garbage collector and
+reducing the cache locality of the program's code.  Spacetime provides
+full backtraces for every allocation that occurred on the OCaml heap
+during the lifetime of the program including those in C stubs.
+
+Spacetime only analyses the memory behaviour of a program with respect to
+the OCaml heap allocators and garbage collector.  It does not analyse
+allocation on the C heap.  Spacetime does not affect the memory behaviour
+of a program being profiled with the exception of any change caused by the
+overhead of profiling (see section\ \ref{runtimeoverhead})---for example
+the program running slower might cause it to allocate less memory in total.
+
+Spacetime is currently only available for x86-64 targets and has only been
+tested on Unix-like systems.  It is expected that supported platforms will
+be extended in the future.
+
+\section{How to use it}
+
+\subsection{Building}
+
+To use Spacetime it is necessary to use an OCaml compiler that was
+configured with the {\tt -spacetime} option.  It is not possible to select
+Spacetime on a per-source-file basis or for a subset of files in a project;
+all files involved in the executable being profiled must be built with the
+Spacetime compiler.
+
+If the {\tt libunwind} library is not available on the system then it will
+not be possible for Spacetime to profile allocations occurring within
+C stubs.  If the {\tt libunwind} library is available but in an unusual
+location then that location may be specified to the {\tt configure} script
+using the {\tt -libunwinddir} option (or alternatively, using separate
+{\tt -libunwindinclude} and {\tt -libunwindlib} options).
+
+OPAM switches will be provided for Spacetime-configured compilers.
```

Comment:
This doesn't really tell the user anything useful (and is future looking since the OPAM switches don't exist yet).